fetch_response.go

```diff
@@ -450,7 +452,15 @@ func (r *FetchResponse) AddRecordWithTimestamp(topic string, partition int32, ke
 		frb.RecordsSet = []*Records{&records}
 	}
 	batch := frb.RecordsSet[0].RecordBatch
-	rec := &Record{Key: kb, Value: vb, OffsetDelta: offset, TimestampDelta: timestamp.Sub(batch.FirstTimestamp)}
```

Comment:
Fundamentally the issue here is that it needs to be possible to include headers in the record, not these specific ones?